### PR TITLE
Removed "Facebook" prefix form all classes

### LINKED ISSUE #609: Remove the Facebook prefix from nearly all classes
Point 10 in #587

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #609.

I've used PHPStorms "rename class" functionality to rename all the classes named Facebook*.

PS. Sorry fo an another massive PR.

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -24,9 +24,9 @@
 namespace Facebook;
 
 use Facebook\Authentication\AccessToken;
-use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Exceptions\SDKException;
 
-class FacebookApp implements \Serializable
+class App implements \Serializable
```
Comment (yguedidi) [inline src/App.php]: as this PR is about renaming, what about naming this one `Application`?

File: src/PersistentData/MemoryPersistentDataHandler.php [hunk 2]
```diff
@@ -24,11 +24,9 @@
 namespace Facebook\PersistentData;
 
 /**
- * Class FacebookMemoryPersistentDataHandler
- *
  * @package Facebook
  */
-class FacebookMemoryPersistentDataHandler implements PersistentDataInterface
+class MemoryPersistentDataHandler implements PersistentDataInterface
```
Comment (yguedidi) [inline src/PersistentData/MemoryPersistentDataHandler.php]: what about `InMemoryPersistentDataHandler` here? I feel it's more common

File: tests/AppTest.php [hunk 3]
```diff
@@ -23,20 +23,20 @@
  */
 namespace Facebook\Tests;
 
-use Facebook\FacebookApp;
+use Facebook\App;
 use Facebook\Authentication\AccessToken;
 use PHPUnit\Framework\TestCase;
 
-class FacebookAppTest extends TestCase
+class AppTest extends TestCase
```
Comment (yguedidi) [inline tests/AppTest.php]: then `ApplicationTest` there if you agree on the tested class name :)

File: tests/PersistentData/MemoryPersistentDataHandlerTest.php [hunk 4]
```diff
@@ -23,14 +23,14 @@
  */
 namespace Facebook\Tests\PersistentData;
 
-use Facebook\PersistentData\FacebookMemoryPersistentDataHandler;
+use Facebook\PersistentData\MemoryPersistentDataHandler;
 use PHPUnit\Framework\TestCase;
 
-class FacebookMemoryPersistentDataHandlerTest extends TestCase
+class MemoryPersistentDataHandlerTest extends TestCase
```
Comment (yguedidi) [inline tests/PersistentData/MemoryPersistentDataHandlerTest.php]: then `InMemoryPersistentDataHandlerTest` here

File: src/App.php (hunk 1 above)
Comment (Nyholm) [reply src/App.php]: Thank you

**Comment by Nyholm** [comment]
Thank you for the review. I've fixed your comments and I've updated the docs

**Comment by yguedidi** [comment]
@SammyK can youhave a look at this PR please? thanks

**Comment by Nyholm** [comment]
I've aligned the doc blocks and made sure they follow PSR5 with some help from PHPcs fixer.

**Comment by yguedidi** [comment]
Thanks a lot for PHPCSFixer!!!! Looks like I merged a PR that create lots of conflicts with yours... Can you please rebase on latest master?

**Comment by Nyholm** [comment]
The PR is rebased..

**Comment by Nyholm** [comment]
Oh... new conflicts.. Rebasing again =)

**Comment by Nyholm** [comment]
There, I've rebased again. Please have a look

File: src/Application.php [hunk 5]
```diff
@@ -87,7 +86,7 @@ public function getAccessToken()
     }
 
     /**
-     * Serializes the FacebookApp entity as a string.
+     * Serializes the App entity as a string.
```
Comment (yguedidi) [inline src/Application.php]: Application

File: src/Application.php [hunk 6]
```diff
@@ -97,7 +96,7 @@ public function serialize()
     }
 
     /**
-     * Unserializes a string as a FacebookApp entity.
+     * Unserializes a string as a App entity.
```
Comment (yguedidi) [inline src/Application.php]: Application

File: src/Authentication/OAuth2Client.php [hunk 7]
```diff
@@ -44,16 +41,16 @@ class OAuth2Client
     const BASE_AUTHORIZATION_URL = 'https://www.facebook.com';
 
     /**
-     * The FacebookApp entity.
+     * The App entity.
```
Comment (yguedidi) [inline src/Authentication/OAuth2Client.php]: Application

File: src/BatchRequest.php [hunk 8]
```diff
@@ -114,26 +112,26 @@ public function add($request, $options = null)
     }
 
     /**
-     * Ensures that the FacebookApp and access token fall back when missing.
+     * Ensures that the App and access token fall back when missing.
```
Comment (yguedidi) [inline src/BatchRequest.php]: Application

File: src/BatchRequest.php [hunk 9]
```diff
@@ -114,26 +112,26 @@ public function add($request, $options = null)
...
+     * @throws SDKException
      */
-    public function addFallbackDefaults(FacebookRequest $request)
+    public function addFallbackDefaults(Request $request)
     {
         if (!$request->getApp()) {
             $app = $this->getApp();
             if (!$app) {
-                throw new FacebookSDKException('Missing FacebookApp on FacebookRequest and no fallback detected on FacebookBatchRequest.');
+                throw new SDKException('Missing App on Request and no fallback detected on BatchRequest.');
```
Comment (yguedidi) [inline src/BatchRequest.php]: Application

File: src/Helper/PageTabHelper.php [hunk 10]
```diff
@@ -19,33 +19,30 @@
...
      */
     protected $pageData;
 
     /**
      * Initialize the helper and process available signed request data.
      *
-     * @param FacebookApp    $app          The FacebookApp entity.
-     * @param FacebookClient $client       The client to make HTTP requests.
-     * @param string         $graphVersion The version of Graph to use.
+     * @param Application $app          the App entity
```
Comment (yguedidi) [inline src/Helper/PageTabHelper.php]: Application

File: src/Helper/SignedRequestFromInputHelper.php [hunk 11]
```diff
@@ -19,31 +19,28 @@
...
 {
     /**
-     * @var SignedRequest|null The SignedRequest entity.
+     * @var null|SignedRequest the SignedRequest entity
      */
     protected $signedRequest;
 
     /**
-     * @var FacebookApp The FacebookApp entity.
+     * @var Application the App entity
```
Comment (yguedidi) [inline src/Helper/SignedRequestFromInputHelper.php]: Application

File: src/Helper/SignedRequestFromInputHelper.php [hunk 12]
```diff
@@ -55,11 +52,11 @@
     /**
      * Initialize the helper and process available signed request data.
      *
-     * @param FacebookApp    $app          The FacebookApp entity.
-     * @param FacebookClient $client       The client to make HTTP requests.
-     * @param string         $graphVersion The version of Graph to use.
+     * @param Application $app          the App entity
```
Comment (yguedidi) [inline src/Helper/SignedRequestFromInputHelper.php]: Application

**Comment by yguedidi** [comment]
I'm really sorry @Nyholm, I did a mistake when merging 5.x into master, so tests were broken (looks like I'm tired..), so you will need to rebase again, sorry

**Comment by Nyholm** [comment]
Updated!

**Comment by Nyholm** [comment]
Wohoo tests are green!

**Comment by yguedidi** [comment]
Thanks a lot @Nyholm!!

**Comment by Nyholm** [comment]
Awesome. Thanks!

**Pull request merged**